ui/plugin-system/src/runtime/TimeRangeProvider/query-params.ts

```diff
@@ -157,3 +163,87 @@ export function useSetTimeRangeParams(
   }
   return { timeRange: initialTimeRange, setTimeRange: setTimeRange };
 }
+
+/**
+ * Gets the initial refresh interval taking into account URL params and dashboard JSON duration
+ * Sets refresh query param if it is empty on page load
+ */
+export function useInitialRefreshInterval(dashboardDuration: DurationString): RelativeTimeRange {
+  const [query] = useQueryParams(refreshIntervalQueryConfig, { updateType: 'replaceIn' });
+  const { refresh } = query;
+  return useMemo(() => {
+    let initialTimeRange: RelativeTimeRange = { pastDuration: dashboardDuration };
+    if (!refresh) {
+      return initialTimeRange;
+    }
+    const startStr = refresh.toString();
+    if (isDurationString(startStr)) {
+      initialTimeRange = { pastDuration: startStr };
+    }
+    return initialTimeRange;
+  }, [dashboardDuration, refresh]);
+}
+
+function getRefreshIntervalInMs(refreshInterval: RelativeTimeRange | undefined) {
```

Comment:
```suggestion
function getRefreshIntervalInMs(refreshInterval?: RelativeTimeRange) {
```

Reply:
You might want to remake these changes locally, accepting suggestions can make DCO start failing